TCC.Lib/Helpers/FileExtensions.cs

```diff
@@ -15,13 +49,13 @@ public static void CreateEmptyFile(this string filePath)
             }
         }
 
-        public static void TryDeleteFileWithRetry(this string filePath, int retries = 100)
+        public static async Task TryDeleteFileWithRetry(this string filePath, int retries = 100)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this method into two, one handling parameters check and the other handling the asynchronous code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS4457)